src/typings/generators.d.ts

```diff
@@ -232,6 +232,45 @@ interface GeneratedFile {
   extension: string
 }
 
+interface PageFactoryParams {
+  componentGenerator: ComponentGenerator
+  stateBranch: StateBranch
+  routerDefinitions: StateDefinition
+  componentOptions: {
+    assetsPrefix: string
+    localDependenciesPrefix: string
+  }
+  pageMetadataOptions?: {
+    usePathAsFileName?: boolean
+    convertDefaultToIndex?: boolean
+  }
+  pageExtension?: string
+}
+
+interface ComponentFactoryParams {
+  componentGenerator: ComponentGenerator
+  componentUIDL: ComponentUIDL
+  componentOptions: {
+    assetsPrefix: string
+    localDependenciesPrefix?: string
+  }
+  componentExtension?: string
+}
+
+interface FolderStructureParams {
+  distFolderName: string
```

Comment:
this interface looks like it doesn't scale well, can we think of different strategies for sending all the different files for different folders? Maybe with some sort of folder identifer?